code/datums/traits/_trait.dm

```diff
@@ -48,6 +48,8 @@
 	if(QDELETED(trait_holder))
 		qdel(src)
 		return
+	if(istype(src, /mob/living/carbon/human/dummy))
```

Comment:
Probably want to check `trait_holder` here, not `src`.

Reply:
You are absolutely right!